Sources/XCResultKit/Schema/TestExpression.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  TestExpression.swift
+//  
+//
+//  Created by Blechman, Ben on 6/24/24.
+//
+
+import Foundation
+
+// Note that because this type uses nested recursion, it cannot be a struct, it must be a class
+//- TestExpression
+//  * Kind: object
+//  * Properties:
+//	+ sourceCode: String
+//	+ value: TestValue?
+//	+ subexpressions: [TestExpression]
+public class TestExpression: XCResultObject {
```

Comment:
do you have any reason to use `class` instead of `struct` ?

Reply:
What I found was that if the schema calls for nested recursion (i.e. there are references to the same type as a member variable), the swift compiler complains about using struct as it s a value type.  So class was the only solution I found.

Reply:
Note, I found it somewhat inconsistent when the compiler complains about this.  It does complain in my project where XCResultKit is a dependency, but not in the XCResultKit project when opened alone.  My guess is it's a difference in the swift tools version (my project is newer).